# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/shadow/hotfix-cpp-decls.patch
# Copyright (C) 2023 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

--- shadow-4.14.1/lib/readpassphrase.h.vanilla	2023-10-27 16:32:27.538774505 +0200
+++ shadow-4.14.1/lib/readpassphrase.h	2023-10-27 16:35:07.892790990 +0200
@@ -36,8 +36,12 @@
 #endif
 #include <sys/types.h>
 
-__BEGIN_DECLS
+#ifdef __cplusplus
+extern "C" {
+#endif
 char * readpassphrase(const char *, char *, size_t, int);
-__END_DECLS
+#ifdef __cplusplus
+}
+#endif
 
 #endif /* !LIBBSD_READPASSPHRASE_H */
